File: PortalOS/PortalOS.Domain/Entities/OrdemServico.cs
using dNET.Domain.Entities;

namespace PortalOS.Domain.Entities
{
    public class OrdemServico : EntidadeBase
    {
        public Projeto Projeto { get; set; }

        public DateTime DataAgenda { get; set; }

        public DateTime HoraInicio { get; set; }

        public DateTime? InicioIntervalo { get; set; }

        public DateTime? FimIntervalo { get; set; }

        public DateTime HoraFim { get; set; }

        public string Descricao { get; set; }

        public string Colaborador { get; set; }
    }
}

File: PortalOS/PortalOS.Web/src/types/ordemServico.ts
export interface OrdemServico {
  id: number;
  tarefaId: number;
  tarefaNome: string;
  projetoId: number;
  projetoNome: string;
  clienteId: number;
  clienteNome: string;
  dataAgenda: string;
  horaInicio: string;
  inicioIntervalo: string | null;
  fimIntervalo: string | null;
  horaFim: string;
  descricao: string;
  colaboradorId: number | null;
  colaboradorNome: string;
  totalHoras: number;
}

export interface CreateOrdemServicoRequest {
  tarefaId: number;
  horaInicio: string;
  inicioIntervalo?: string | null;
  fimIntervalo?: string | null;
  horaFim: string;
  descricao: string;
}

export interface UpdateOrdemServicoRequest {
  id: number;
  tarefaId: number;
  dataAgenda: string;
  horaInicio: string;
  inicioIntervalo?: string | null;
  fimIntervalo?: string | null;
  horaFim: string;
  descricao: string;
}
